### PR TITLE
perf(ingest): changes to improve ingest performance a bit

### PR DESCRIPTION
**Pull request by anshbansal**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java [hunk 1]
```diff
@@ -233,7 +237,46 @@ public abstract VersionedAspect getVersionedAspect(@Nonnull final Urn urn, @Nonn
...
+  /**
+   * Same as ingestAspectToLocalDB but for multiple aspects
+   */
+  @Nonnull
+  protected abstract List<Pair<String, UpdateAspectResult>> ingestAspectsToLocalDB(@Nonnull final Urn urn,
+    @Nonnull List<Pair<String, RecordTemplate>> aspectRecordsToIngest,
+    @Nonnull final AuditStamp auditStamp, @Nonnull final SystemMetadata providedSystemMetadata);
+
+  @Nonnull
+  private SystemMetadata generateSystemMetadataIfEmpty(SystemMetadata systemMetadata) {
```
Comment (jjoyce0510) [inline metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java:251]: One minor nit: Typically we place private methods below public methods

Comment (jjoyce0510) [reply metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java:251]: (And typically we place callee below caller methods)

File: metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java [hunk 2]
```diff
@@ -233,7 +237,46 @@ public abstract VersionedAspect getVersionedAspect(@Nonnull final Urn urn, @Nonn
...
+    }
+  }
+
+  public void ingestAspects(@Nonnull final Urn urn, @Nonnull List<Pair<String, RecordTemplate>> aspectRecordsToIngest,
+    @Nonnull final AuditStamp auditStamp, SystemMetadata systemMetadata) {
+
+    validateUrn(urn);
+    systemMetadata = generateSystemMetadataIfEmpty(systemMetadata);
+
+    Timer.Context ingestToLocalDBTimer = MetricUtils.timer(this.getClass(), "ingestAspectsToLocalDB").time();
```
Comment (jjoyce0510) [inline metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java:272]: Nit: should this be ingestAspects?

File: metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java [hunk 3]
```diff
@@ -586,20 +616,44 @@ public void ingestEntity(@Nonnull Entity entity, @Nonnull AuditStamp auditStamp,
             .collect(Collectors.toList())));
   }
 
+  /**
+  Returns true if entityType should have some aspect as per its definition
```
Comment (jjoyce0510) [inline metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java:620]: This is still somewhat misleading in the fact that an entity may have an aspect like "Ownership" but it's not always _required_ to be provided in the set of aspects

Comment (jjoyce0510) [reply metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java:620]: We could call this something like "isAspectProvided"

File: metadata-io/src/main/java/com/linkedin/metadata/entity/ebean/EbeanEntityService.java [hunk 4]
```diff
@@ -274,44 +309,82 @@ protected UpdateAspectResult ingestAspectToLocalDB(@Nonnull final Urn urn, @Nonn
...
-        SystemMetadata latestSystemMetadata = EbeanUtils.parseSystemMetadata(latest.getSystemMetadata());
-        latestSystemMetadata.setLastObserved(providedSystemMetadata.getLastObserved());
+  @Override
+  @Nonnull
+  protected List<Pair<String, UpdateAspectResult>> ingestAspectsToLocalDB(@Nonnull final Urn urn,
+     @Nonnull List<Pair<String, RecordTemplate>> aspectRecordsToIngest,
+     @Nonnull final AuditStamp auditStamp, @Nonnull final SystemMetadata systemMetadata) {
 
-        latest.setSystemMetadata(RecordUtils.toJsonString(latestSystemMetadata));
+    return _entityDao.runInTransactionWithRetry(() -> {
```
Comment (jjoyce0510) [inline metadata-io/src/main/java/com/linkedin/metadata/entity/ebean/EbeanEntityService.java:326]: So this block will better batch these operations to MySQL?

**Review by jjoyce0510 — APPROVED** [review]
Overall LGTM. Let's monitor this one closely since it's a fairly significant change

File: metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java (hunk 2 above)
Comment (anshbansal) [reply metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java:272]: No we are measuring what is running inside the timer bounds which is `ingestAspectsToLocalDB` method

File: metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java (hunk 3 above)
Comment (anshbansal) [reply metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java:620]: will change to that

**Pull request closed without merging**